lib/tests/streamlit/elements/data_editor_test.py

```diff
@@ -57,9 +59,79 @@ def _get_arrow_schema(df: pd.DataFrame) -> pa.Schema:
 
 
 class DataEditorUtilTest(unittest.TestCase):
-    def test_parse_value(self):
-        # TODO: test parse_value
-        pass
+    @parameterized.expand(
+        [
+            (None, ColumnDataKind.STRING, None),
+            ("hello", ColumnDataKind.STRING, "hello"),
+            (123, ColumnDataKind.STRING, "123"),
+            (None, ColumnDataKind.INTEGER, None),
+            ("123", ColumnDataKind.INTEGER, 123),
+            (123, ColumnDataKind.INTEGER, 123),
+            (123.1234, ColumnDataKind.INTEGER, 123),
+            (None, ColumnDataKind.FLOAT, None),
+            ("123.45", ColumnDataKind.FLOAT, 123.45),
+            (123.45, ColumnDataKind.FLOAT, 123.45),
+            (123, ColumnDataKind.FLOAT, 123),
+            (None, ColumnDataKind.BOOLEAN, None),
+            (True, ColumnDataKind.BOOLEAN, True),
+            ("true", ColumnDataKind.BOOLEAN, True),
+            (None, ColumnDataKind.DATETIME, None),
+            (
```

Comment:
Should we consider having none with the datetime too?

Reply:
We have one `None` check for datetime above: `(None, ColumnDataKind.DATETIME, None),`, I also added additional ones for time and date 👍